tools/wake-format/emitter.cpp

```diff
@@ -619,7 +649,35 @@ wcl::doc Emitter::walk_package(ctx_t ctx, CSTElement node) {
 
 wcl::doc Emitter::walk_paren(ctx_t ctx, CSTElement node) {
   MEMO(ctx, node);
-  MEMO_RET(walk_placeholder(ctx, node));
+  assert(node.id() == CST_PAREN);
+
+  // clang-format off
+  auto comment_fmt =
+      fmt().nest(fmt()
+             .token(TOKEN_P_POPEN)
+             .fmt_if(has_comment_next, position_comment().join(fmt().token(TOKEN_COMMENT)))
```

Comment:
I think the `join(fmt().token(TOKEN_COMMENT)))` can be dropped perhaps? You'll have to make comments have trailing newlines instead of proceeding newlines.

Reply:
we talked about it in DM, posting here for posterity

Moving/deleting it would be ideal but that causes an extra trailing newline right before the expression and I'm trying very hard to not use  builder->undo so we have to keep it like this to not generate the extra newline